### PR TITLE
Convert all default exports to named exports

### PR DESCRIPTION
**Pull request by ghoshkaj**
In https://github.com/microsoft/FluidFramework/pull/11851 I investigated whether enforcing named exports would have any effect on bundle sizes and I found that they did not. This PR is to convert all existing default exports into named exports.

Related issue: https://github.com/microsoft/FluidFramework/issues/10062

### CODE REVIEW

File: packages/drivers/local-driver/src/localDocumentServiceFactory.ts [hunk 1]
```diff
@@ -76,6 +76,7 @@ export class LocalDocumentServiceFactory implements IDocumentServiceFactory {
             defaultHash,
             resolvedUrl.endpoints.ordererUrl ?? "",
             resolvedUrl.endpoints.storageUrl ?? "",
+            "",
```
Comment (ChumpChief) [inline packages/drivers/local-driver/src/localDocumentServiceFactory.ts:79]: This seems like an error?

**Review by tylerbutler — CHANGES_REQUESTED** [review]
Please don't add new code to the @fluidframework/build-tools package. Instead, new code should be put in the @fluid-tools/build-cli package.

Since you're writing a command line tool, you should use [oclif](https://oclif.io/docs/introduction) which we use for the rest of our build tools (typetests are moving there too). A good example/starting point for a command is the info command: https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/build-cli/src/commands/info.ts

oclif and our shared BaseCommand class has some helpful things, including automatic argument parsing and validation, consistent logging, and async APIs.

You can also add unit tests for any functions, and you can add tests for the CLI itself. See an example here: https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/build-cli/test/commands/generate/buildVersion.test.ts

All that said, could this also be implemented as a policy handler? Here's an example: https://github.com/microsoft/FluidFramework/blob/main/build-tools/packages/build-tools/src/repoPolicyCheck/handlers/lockfiles.ts

(Those files are still in the old build-tools project but are being moved.)

File: packages/drivers/local-driver/src/localDocumentServiceFactory.ts (hunk 1 above)
Comment (ghoshkaj) [reply packages/drivers/local-driver/src/localDocumentServiceFactory.ts:79]: Yeah my package versions seem to be a bit off so I'm looking into fixing these errors. I will remove this!

**Comment by ghoshkaj** [comment]
@ChumpChief @tylerbutler I am opening a new PR that will have cleaner syntax. I will link to it here once it's up and ready.

@tylerbutler yes I'm not going to be checking in the script in the end. I just checed it in for now to have a record of it. I will still check it into the new PR and remove it, again to have a record of it. For enforcement, I was going to add non default exports as part of the eslint rules.

**Comment by ghoshkaj** [comment]
@ChumpChief @tylerbutler please take a look at the new PR that build with no failures and doesn't check in the explicit export script: https://github.com/microsoft/FluidFramework/pull/12321

**Pull request closed without merging**